### ISSUE TITLE
panic unmarshaling array of tables

### ISSUE THREAD

**Issue by jmank88** [pelletier/go-toml#839]
**Describe the bug**
Unmarshaling to a field of a table in an array of tables causes a panic.

**To Reproduce**
https://go.dev/play/p/l2KDjY4lX-3
```go
m := struct {
	A []struct {
		B struct{}
	}
}{}
fmt.Println(toml.Unmarshal([]byte(`[[A.B]]`), &m))
```

**Expected behavior**
If this is invalid, I expect an error rather than a panic.

**Versions**
 - go-toml: 2.0.6
 - go: 1.19
 - operating system: Linux, Go-playground

**Comment by pelletier** [comment]
Thank you for the bug report! I'll take a look when I have time, but feel free to send a pull request in the meantime!

**Comment by jmank88** [comment]
@pelletier I started a draft that highlights the problematic area (#840). Is returning an error the correct thing to do here?

**Comment by pelletier** [comment]
You're correct: this should be an error, because the array of tables is on `A.B`, and `B` is a struct, not a slice.

**Comment by pelletier** [comment]
Should be fixed now.
